src/data/storage.js

```diff
@@ -15,21 +17,25 @@ export const LIFESPAN = Object.freeze({
   LONG,
 })
 
-export const save = (key, value, LIFESPAN) => LIFESPAN.setItem(key, value)
+export const save = (key, value, lifespan) => {
+  Object.keys(LIFESPAN).filter(type => LIFESPAN[type] !== lifespan).map(type => {
+    const valueFromOtherLocation = LIFESPAN[type].getItem(key)
 
-export function retrieve(key) {
-  let item = LIFESPAN.SHORT.getItem(key)
+    if (valueFromOtherLocation !== null) {
+      console.error(`Storage key name clash:  ${key} found in ${type} already, moving to requested location.`)
+      clear(key)
+    }
+  })
 
-  if (item === null) {
-    item = LIFESPAN.LONG.getItem(key)
-  }
+  lifespan.setItem(key, value)
+}
 
-  return item
+export function retrieve(key) {
+  const values = Object.keys(LIFESPAN).map(type => LIFESPAN[type].getItem(key)).filter(value => value !== null)
```

Comment:
Had trouble reading what was happening here. Perhaps another method could be extracted to hide part of the logic.

Reply:
understandable:  https://github.com/silinternational/wecarry-ui/commit/05cf86ba252fd1d57f8424915148987f15ad3419